trin-core/src/jsonrpc/service.rs

```diff
@@ -171,11 +171,13 @@ fn serve_http_client(
     infura_url: &str,
     portal_tx: UnboundedSender<PortalJsonRpcRequest>,
 ) {
-    let mut buffer = Vec::new();
-    stream.read_to_end(&mut buffer).unwrap();
+    let mut reader = io::BufReader::new(&mut stream);
+    let received: Vec<u8> = reader.fill_buf().unwrap().to_vec();
+    // Mark the bytes read as consumed so the buffer will not return them in a subsequent read
+    reader.consume(received.len());
```

Comment:
Hm, I don't think this is sufficient. `fill_buf()` doesn't seem to make any guarantees about getting a complete message. It could return in the middle. Maybe `read_to_end()` still makes sense here?

Reply:
The problem I was running into with `read_to_end()` is that it hangs until it receives an EOF. Which is not the desired behavior if I'm understanding things correctly. In fact, I'm surprised it was ever functional in the first place.... I do understand the concern with incomplete messages. Do you think the parsing/handling invalid messages in `parse_http_request()` is sufficient? I don't think that we'll expect very large requests that often

Reply:
Hm, I thought that the deal was that a connection is established, then an http request is sent, then the connection exits. That's why I thought reading to EOF would work. That would not for persistent connections, of course, but I didn't think we needed any support for that. I'll admit plenty of ignorance about the details, though.